pkg/provisioner/ironic/clients/client.go

```diff
@@ -26,11 +49,14 @@ func IronicClient(ironicEndpoint string, auth AuthConfig) (client *gophercloud.S
 	default:
 		err = fmt.Errorf("Unknown auth type %s", auth.Type)
 	}
-	return
+	if err != nil {
+		return
+	}
+	return updateHTTPClient(client, caFile)
 }
 
 // InspectorClient creates a client for Ironic Inspector
-func InspectorClient(inspectorEndpoint string, auth AuthConfig) (client *gophercloud.ServiceClient, err error) {
+func InspectorClient(inspectorEndpoint string, auth AuthConfig, caFile string) (client *gophercloud.ServiceClient, err error) {
```

Comment:
Why not make this part of the AuthConfig struct?

Reply:
good point, I will update

Reply:
actually, it's handled quite differently, and the path to the CA is unrelated to the authConfig, so I think keeping it separated might be cleaner/clearer. What do you think ?

Reply:
I missed that this is just TLS, not mTLS, so it's only auth if you squint (technically we are authenticating the server!). It probably does make sense to keep it separate if:

* There are no other options likely to be added (e.g. use https but allow self-signed certs); and
* When mTLS support is added, this remains completely independent (they don't e.g. need to share a CA)

Reply:
I'm playing with the idea of making it a TLSConfig. I'll modify the PR with that, please take a look.